load.php

```diff
@@ -108,7 +108,23 @@ function perflab_get_module_settings() {
 	// Even though a default value is registered for this setting, the default must be explicitly
 	// passed here, to support scenarios where this function is called before the 'init' action,
 	// for example when loading the active modules.
-	return (array) get_option( PERFLAB_MODULES_SETTING, perflab_get_modules_setting_default() );
+	$perflab_get_module_settings = (array) get_option( PERFLAB_MODULES_SETTING, perflab_get_modules_setting_default() );
+
+	$legacy_module_slug = array(
+		'site-health/audit-autoloaded-options' => 'database/audit-autoloaded-options',
+		'site-health/audit-enqueued-assets'    => 'css-and-js/audit-enqueued-assets',
+		'site-health/audit-full-page-cache'    => 'object-cache/audit-full-page-cache',
+		'site-health/webp-support'             => 'images/webp-support',
+	);
+
+	foreach ( $perflab_get_module_settings as $module_slug => $module_data ) {
+		if ( isset( $legacy_module_slug[ $module_slug ] ) ) {
+			$perflab_get_module_settings[ $legacy_module_slug[ $module_slug ] ] = $module_data;
+			unset( $perflab_get_module_settings[ $module_slug ] );
+		}
+	}
+
+	return $perflab_get_module_settings;
```

Comment:
Instead of looping through all the module settings, let's loop through just the legacy module slugs, as that is more efficient. It also makes the code easier to understand.
```suggestion
	foreach ( $legacy_module_slugs as $legacy_slug => $current_slug ) {
		if ( isset( $module_settings[ $legacy_slug ] ) ) {
			$module_settings[ $current_slug ] = $module_settings[ $legacy_slug ];
			unset( $module_settings[ $legacy_slug ] );
		}
	}

	return $module_settings;
```

Reply:
Thanks for the review. This change make more sense.